### PR TITLE
[Feature] Support export without broker

### LINKED ISSUE #9170: Support export without broker

### PR DESCRIPTION
**Pull request by xiaoyong-z**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9170

## Problem Summary(Required) ：

Support export without broker

### CODE REVIEW

File: be/src/fs/fs_hdfs.cpp [hunk 1]
```diff
@@ -194,8 +268,96 @@ class HdfsFileSystem : public FileSystem {
...
+    FSOptions _options;
 };
 
+Status HdfsFileSystem::path_exists(const std::string& path) {
+    std::string namenode;
+    RETURN_IF_ERROR(get_namenode_from_path(path, &namenode));
+    HdfsFsHandle handle;
+    RETURN_IF_ERROR(HdfsFsCache::instance()->get_connection(namenode, &handle, _options));
+    if (handle.type != HdfsFsHandle::Type::HDFS) {
+        return Status::InvalidArgument("invalid hdfs path");
```
Comment (chaoyli) [inline be/src/fs/fs_hdfs.cpp:284]: add path in the error status

File: be/src/fs/fs_hdfs.cpp [hunk 2]
```diff
@@ -194,8 +268,96 @@ class HdfsFileSystem : public FileSystem {
...
+    }
+
+    int flags = O_WRONLY;
+    if (opts.mode == FileSystem::CREATE_OR_OPEN_WITH_TRUNCATE) {
+        if (auto st = _path_exists(handle.hdfs_fs, path); st.ok()) {
+            return Status::NotSupported("Cannot truncate a file by hdfs writer");
```
Comment (chaoyli) [inline be/src/fs/fs_hdfs.cpp:311]: add the path in the error status

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 3]
```diff
@@ -1368,6 +1368,14 @@ opt_broker ::=
     {:
         RESULT = null;
     :}
+    | KW_WITH KW_BROKER
```
Comment (chaoyli) [inline fe/fe-core/src/main/cup/sql_parser.cup:1371]: KW_WITH HDFS/S3 ?

Comment (xiaoyong-z) [reply fe/fe-core/src/main/cup/sql_parser.cup]: We can infer HDFS or S3  from the uri of the file, no need to add HDFS/S3 here

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 40 (10.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|12|00.00%|[354, 355, 356, 631, 632, 634, 643, 644, 646, 666, 667, 669]|
|:large_blue_circle:|com/starrocks/planner/ExportSink.java|0|7|00.00%|[47, 53, 59, 96, 97, 98, 99]|
|:large_blue_circle:|com/starrocks/task/ExportExportingTask.java|0|14|00.00%|[226, 227, 228, 229, 230, 231, 234, 235, 236, 237, 238, 244, 245, 247]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|1|00.00%|[851]|
|:large_blue_circle:|com/starrocks/analysis/ExportStmt.java|4|6|66.67%|[199, 204]|

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**